.github/workflows/go.yml

```diff
@@ -1,21 +1,24 @@
 name: Build and Test
 on: [push, pull_request]
+permissions:
+  contents: read
```

Comment:
What does this do?

Reply:
I believe just to make sure actions can't change the repo?  I am not sure who added it first, but it's just being copied around to keep the go.yml's mostly in-sync :)

https://docs.github.com/en/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-contents